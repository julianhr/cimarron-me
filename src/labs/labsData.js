import React from 'react'


const data = [
  {
    title: 'Infinite Scroller',
    component: React.lazy(
      () => import(/* webpackChunkName: "InfiniteScroller" */ '~/labs/001_InfiniteScroller')
    ),
    urlPath: '/labs/infinite-scroller',
    thumbnailSrc: require('../labs/001_InfiniteScroller/thumbnail.png'),
    stack: 'React.js Javascript'.split(' '),
    createdAt: '2019-03',
  }
]

export default data
