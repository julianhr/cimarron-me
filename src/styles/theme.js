export const colors = {
  background: '#FFF',
  code: '#D90000',
  error: '#d400ff',
  field: {
    background: '#f7f7f7',
    border: '#BBB',
  },
  highlight: '#FFD342',
  text: '#222222',
  primary: {
    dark: '#9B1518',
    light: '#CE1F23',
    dim: '#FFA8AA',
  },
}

export const breaks = {
  xs: 0,
  sm: 420,
  md: 900,
  lg: 1200,
}

export const breakBump = {
  xs: 'sm',
  sm: 'md',
  md: 'lg',
}

export const queries = {
  xs: `@media (max-width: ${breaks.sm - 1}px)`,
  sm: `@media (min-width: ${breaks.sm}px) and (max-width: ${breaks.md - 1}px)`,
  md: `@media (min-width: ${breaks.md}px) and (max-width: ${breaks.lg - 1}px)`,
  lg: `@media (min-width: ${breaks.lg}px)`,
  from: (breakPoint) => (`@media (min-width: ${breaks[breakPoint]}px)`),
  upTo: (breakPoint) => (`@media (max-width: ${breaks[ breakBump[breakPoint] ] - 1}px)`),
}

const theme = {
  colors,
  breaks,
  queries,
}

export default theme
