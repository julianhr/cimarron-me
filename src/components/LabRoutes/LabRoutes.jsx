import React, { Suspense } from 'react'
import PropTypes from 'prop-types'
import { Switch, Route } from 'react-router-dom'
import { connect } from 'react-redux'

import labs from '~/labs/labsData'
import LabShowcase from '../LabShowcase/LabShowcase'
import Loading from '../Loading/Loading'


function LabRoutes({ labs }) {
  const getRoutes = () => (
    labs.map(lab => (
      <Route
        key={lab.urlPath}
        path={`/${lab.urlPath}`}
        component={lab.component}
      />
    ))
  )

  return (
    <Suspense fallback={<Loading />}>
      <Switch>
        <Route exact path='/' component={LabShowcase} />
        {getRoutes()}
      </Switch>
    </Suspense>
  )
}

LabRoutes.propTypes = {
  labs: PropTypes.arrayOf(PropTypes.object),
  pathname: PropTypes.string,
}

const mapStateToProps = () => ({ labs })

export default connect(mapStateToProps)(LabRoutes)
