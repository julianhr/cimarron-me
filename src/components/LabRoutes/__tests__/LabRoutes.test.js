import React from 'react'
import { create } from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from 'emotion-theming'
import { shallow, render } from 'enzyme'

import theme from '~/styles/theme'
import LabRoutes from '../LabRoutes'
import labsData from '~/__tests__/__fixtures__/labsDataFixture'
import TestCompSync from '~/__tests__/__fixtures__/TestComp2'


function TestLabRoutes({ labs, path }) {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={[path]}>
        <LabRoutes labs={labs} />
      </MemoryRouter>
    </ThemeProvider>
  )
}

describe('LabRoutes', () => {
  describe('snapshots', () => {
    it('route to Home', () => {
      const root = create(<TestLabRoutes labs={labsData} path={'/'} />)
      expect(root).toMatchSnapshot()
    })

    // React.lazy is not supported yet since Jest relies on SSR
    // https://github.com/facebook/react/pull/14626#issuecomment-458959723
    // https://github.com/airbnb/enzyme/pull/1975
    // TODO: revisit at a later point
    it.skip('route to component resolved by Suspense', async () => {
      const lab1 = labsData[0]
      lab1.component = React.lazy(() => import('~/__tests__/__fixtures__/TestComp1'))
      await lab1.component
      const wrapper = create(<TestLabRoutes labs={labsData} path={lab1.urlPath} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('loading state', async () => {
      const lab1 = labsData[0]
      lab1.component = React.lazy(() => new Promise(resolve => setTimeout(() => TestCompSync, 500)))
      const wrapper = create(<TestLabRoutes labs={labsData} path={lab1.urlPath} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
