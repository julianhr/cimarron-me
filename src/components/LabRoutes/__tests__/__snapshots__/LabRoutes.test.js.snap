// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabRoutes snapshots loading state 1`] = `
<div
  className="css-16yj1vq-Root e17hmr3i0"
>
  <div
    className="css-14q6cg7-Container e17hmr3i1"
  >
    <span>
      Loading
    </span>
    <span
      className="css-1ui7ev8-Dot1-spanBase e17hmr3i2"
    >
      .
    </span>
    <span
      className="css-nxj8q0-Dot2-spanBase e17hmr3i3"
    >
      .
    </span>
    <span
      className="css-19ldbnk-Dot3-spanBase e17hmr3i4"
    >
      .
    </span>
  </div>
</div>
`;

exports[`LabRoutes snapshots route to Home 1`] = `
<section
  className="css-1lzrryp-Root e1uula6j0"
>
  <a
    className="css-h8xsv8-StyledLink-baseArticle e1lgusa60"
    href="/path/1"
    onClick={[Function]}
  >
    <article
      className="css-xjgwqh-Article-baseArticle e1lgusa61"
    >
      <figure
        className="css-1t2crv9-Root-ImageLoader e1wlpvq40"
      >
        <img
          className="css-jeu92c-Img-ImageLoader e1wlpvq41"
          onLoad={[Function]}
          src="http://image1.com"
        />
      </figure>
      <h3>
        Title 1
      </h3>
    </article>
    <section
      className="css-1k4ntzt-TechRoot e1lgusa62"
    >
      <div
        className="css-if30yk-Tech e1lgusa63"
      >
        React
      </div>
      <div
        className="css-1h4zuz1-Tech e1lgusa63"
      >
        Python
      </div>
    </section>
  </a>
  <a
    className="css-h8xsv8-StyledLink-baseArticle e1lgusa60"
    href="/path/2"
    onClick={[Function]}
  >
    <article
      className="css-xjgwqh-Article-baseArticle e1lgusa61"
    >
      <figure
        className="css-1t2crv9-Root-ImageLoader e1wlpvq40"
      >
        <img
          className="css-jeu92c-Img-ImageLoader e1wlpvq41"
          onLoad={[Function]}
          src="http://image2.com"
        />
      </figure>
      <h3>
        Title 2
      </h3>
    </article>
    <section
      className="css-1k4ntzt-TechRoot e1lgusa62"
    >
      <div
        className="css-if30yk-Tech e1lgusa63"
      >
        Ruby
      </div>
      <div
        className="css-1h4zuz1-Tech e1lgusa63"
      >
        Go
      </div>
    </section>
  </a>
</section>
`;

exports[`LabRoutes snapshots route to component resolved by Suspense 1`] = `
<component 1
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "r0q6qy",
          "pathname": "/path/1",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "r0q6qy",
        "pathname": "/path/1",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  location={
    Object {
      "hash": "",
      "key": "r0q6qy",
      "pathname": "/path/1",
      "search": "",
      "state": undefined,
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/path/1",
      "url": "/path/1",
    }
  }
/>
`;
